File: base/src/main/java/com/juhezi/module/base/common/SingleFragmentActivity.java
package com.juhezi.module.base.common;

import android.os.Bundle;
import android.support.annotation.IdRes;
import android.support.annotation.LayoutRes;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v7.app.AppCompatActivity;

/**
 * Activity-Single-Fragment Base
 * Created by Juhezi[juhezix@163.com] on 2017/5/24.
 */
public abstract class SingleFragmentActivity extends AppCompatActivity {

    protected abstract Fragment createFragment();

    protected void init(@Nullable Bundle savedInstanceState) {
    }

    /**
     * Activity Layout Resource
     * @return Layout Resource
     */
    @LayoutRes
    protected abstract int getLayoutResId();

    /**
     * Fragment View Wrapper Id
     *
     * @return
     */
    @IdRes
    protected abstract int getFragmentWrapperId();

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(getLayoutResId());
        FragmentManager fm = getSupportFragmentManager();
        Fragment fragment = fm.findFragmentById(getFragmentWrapperId());
        if (null == fragment) {
            fragment = createFragment();
            fm.beginTransaction()
                    .add(getFragmentWrapperId(), fragment)
                    .commit();
        }
        init(savedInstanceState);
    }
}

File: app/src/main/java/com/juhezi/coderslife/SingleFragmentActivity.kt
package com.juhezi.coderslife

import android.os.Bundle
import android.support.annotation.IdRes
import android.support.annotation.LayoutRes
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v7.app.AppCompatActivity
import android.util.Log

/**
 * Created by qiao1 on 2017/1/9.
 */
abstract class SingleFragmentActivity : AppCompatActivity() {

    private var TAG = "SingleFragmentActivity"

    protected open fun init() {
    }

    protected abstract fun getFragment(): Fragment

    @LayoutRes
    protected abstract fun getActLayoutRes(): Int

    @IdRes
    protected abstract fun getFragContainerId(): Int

    protected var mFragment: Fragment? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Log.i(TAG,this.toString())
        setContentView(getActLayoutRes())
        var fragmentManager = supportFragmentManager
        mFragment = fragmentManager.findFragmentById(getFragContainerId())
        if (mFragment == null) {
            mFragment = getFragment()
            fragmentManager.beginTransaction()
                    .add(getFragContainerId(), mFragment)
                    .commit()
        }
        init()
    }

}